api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -138,6 +138,15 @@ type AzureManagedControlPlaneSpec struct {
 	// AutoscalerProfile is the parameters to be applied to the cluster-autoscaler when enabled
 	// +optional
 	AutoScalerProfile *AutoScalerProfile `json:"autoscalerProfile,omitempty"`
+
+	// AzureEnvironment is the name of the AzureCloud to be used.
+	// The default value that would be used by most users is "AzurePublicCloud", other values are:
+	// - ChinaCloud: "AzureChinaCloud"
+	// - GermanCloud: "AzureGermanCloud"
```

Comment:
nit: GermanCloud is deprecated since 2018 and I don't think AKS was ever supported there so we should probably remove it from this list

https://learn.microsoft.com/en-us/azure/germany/